WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -654,16 +662,23 @@ private extension StoreCreationCoordinator {
     func waitForSiteToBecomeJetpackSite(from navigationController: UINavigationController, siteID: Int64, expectedStoreName: String) {
         /// Free trial sites need more waiting time that regular sites.
         ///
-        let retryInterval: UInt64 = isFreeTrialCreation ? 10_000_000_000 : 5_000_000_000
         siteIDFromStoreCreation = siteID
 
         jetpackSiteSubscription = $siteIDFromStoreCreation
             .compactMap { $0 }
             .removeDuplicates()
             .asyncMap { [weak self] siteID -> Site? in
+                guard let self else {
+                    return nil
+                }
                 // Waits some seconds before syncing sites every time.
-                try await Task.sleep(nanoseconds: retryInterval)
-                return try await self?.syncSites(forSiteThatMatchesSiteID: siteID, expectedStoreName: expectedStoreName)
+                do {
+                    try await Task.sleep(nanoseconds: UInt64(self.jetpackCheckRetryInterval * 1_000_000_000))
+                    return try await self.syncSites(forSiteThatMatchesSiteID: siteID, expectedStoreName: expectedStoreName)
+                } catch {
+                    self.storeCreationProgressHostingViewController?.viewModel.incrementProgress()
```

Comment:
💭 for best Rx practices, there should be no side effects in the map logic (updating other states that are not involved in the map). To perform side effects, it's best to use `handleEvents`.

Reply:
Good point. I agree. Done in 9f5f230354a20ce350a0a586e185cda160786a72